sup3r/preprocessing/exogenous_data_handling.py

```diff
@@ -0,0 +1,83 @@
+"""Sup3r exogenous data handling"""
+
+import logging
+import numpy as np
+
+from sup3r.utilities.topo import TopoExtract
+
+
+logger = logging.getLogger(__name__)
+
+
+class ExogenousDataHandler:
+    """Class to extract exogenous features for multistep forward passes. e.g.
+    Multiple topography arrays at different resolutions for multiple spatial
+    enhancement steps."""
+
+    def __init__(self, file_paths, features, source_h5, s_enhancements,
+                 agg_factors, target=None, shape=None, raster_file=None,
+                 max_delta=20, input_handler=None):
+        """
+        Parameters
+        ----------
+        file_paths : str | list
+            A single source h5 wind file to extract raster data from or a list
+            of netcdf files with identical grid. The string can be a unix-style
+            file path which will be passed through glob.glob
+        features : list
+            List of exogenous features to extract from source_h5
+        source_h5 : str
+            Filepath to source wtk or nsrdb file to get hi-res (2km or 4km)
+            data from which will be mapped to the enhanced grid of
+            the file_paths input
+        s_enhancements : list
```

Comment:
So if there's two spatial enhancement steps and a single temporal enhancement step, should s_enhancements be [5, 5, None] or just [5, 5]? Could use clarification on if the length of these args should match the number of models or just the number of spatial steps. I would prefer the former as it is more explicit, but I dont think a None entry is handled below.